src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -496,12 +505,6 @@ def _set_parallel_devices_and_init_accelerator(self) -> None:
             self.accelerator: Accelerator = self._accelerator_flag
         else:
             assert self._accelerator_flag is not None
```

Comment:
Is this assert still relevant?

Reply:
I think so because of the type check in mypy where AcceleratorRegistry.get() does not accept None as input.